build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -211,15 +237,19 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
                     Environment.GetEnvironmentVariable("ProgramFiles(x86)") :
                     Environment.GetEnvironmentVariable("ProgramFiles");
 
-                if (configuration.Equals("Release"))
+                if (arch == "arm64")
                 {
-                    // Cmake calls it "RelWithDebInfo" in the generated MSBuild
-                    configuration = "RelWithDebInfo";
+                    Exec(Path.Combine(pf32, "MSBuild", "14.0", "Bin", "MSBuild.exe"),
+                    Path.Combine(cmakeOut, "ALL_BUILD.vcxproj"),
+                    "/p:useEnv=true",
```

Comment:
Refactor this if/else to extract `/p:useEnv=true` for arm64 so that we have a single piece of code to maintain.